clang/test/CodeGenSYCL/loop_fusion_host.cpp

```diff
@@ -0,0 +1,32 @@
+// RUN: %clang_cc1 -fsycl -fsycl-is-host -triple -x86_64-unknown-linux-gnu -disable-llvm-passes -emit-llvm -o - %s | FileCheck %s
+
+template <typename name, typename Func>
+__attribute__((sycl_kernel)) void kernel(const Func &kernelFunc) {
+  kernelFunc();
+}
+
+template <int SIZE>
+class KernelFunctor5 {
+public:
+  [[intel::loop_fuse(SIZE)]] void operator()() const {}
+};
+
+template <int SIZE>
+class KernelFunctor3 {
+public:
+  [[intel::loop_fuse_independent(SIZE)]] void operator()() const {}
+};
+
+[[intel::loop_fuse]] void func1() {}
+[[intel::loop_fuse_independent]] void func2() {}
+
```

Comment:
Now that I understand this test file a bit better (thank you for the explanation on host vs device!), I think this would be a useful test to add to both the device and host test files:
```
int foo();
[[intel::loop_fuse(foo())]] void whatever();
```
If you add this test to loop_fuse_device.cpp, I bet we'll get a diagnostic about `foo()` not being a valid constant expression, but in this test file, I think it will be silently accepted.

If you don't want to worry about the silent acceptance problem right now, you can add a FIXME comment to the host test and open an issue so we remember to think about the problem for all the attributes where it matters.

Reply:
Oh I see what you mean now! Sorry, my brain was fixating on 'but we ignore the attribute' and being obtuse. You are concerned about invalid code being accepted on host irrespective of whether the attribute is ignored or not. From a codegen point of view, it doesn't matter much because we just ignore the attribute anyway, but I guess compiler shouldn't be accepting invalid code. I think all I need to do is remove this check and conditionalize `addAttr `.  I'll make the change. Thanks!

Reply:
Done!

Reply:
Thanks!